### ISSUE TITLE
use uri-bytestring for urls instead of ST

### ISSUE THREAD

**Issue by fisx** [liqd/aula#31]
Also relevant: http://hackage.haskell.org/package/url-generic-0.1/docs/Web-URL-Generic.html (similar to the idea of `Frontend.Path`; unmaintained as of 2016-02-25)

**Comment by fisx** [comment]
also relevant

**Comment by fisx** [comment]
edited description.

**Comment by fisx** [comment]
I am currently leaning towards writing more documentation for `Data.UriPath` and `Frontend.Path`, and not draw in any more libraries, but only comment on what we are doing differently.

If nobody objects, I will make a PR to that end soonish.

**Comment by fisx** [comment]
More comments from brief off-line discussion with @np:

- It may still be nice to replace `ST` with `uri-bytestring` as a target type in `Data.UriPath`.
- It may be overkill, too, as we are only dealing with paths.  But `uri-bytestring` supports paths, relative refs, and we may still need more of the full URI structure in the future.

**Comment by fisx** [comment]
@np: I just tried to change the avatar field in `User` from `ST` to `Maybe UriPath`, but that would have destroyed `Eq` and `Show` derivability.

What do you think about changing the implementation of `UriPath` to something slower, but more manageable and straight-forward?  (Alternatively, we could just use uri-bytestring here.)

**Comment by fisx** [comment]
[irc]

```
10:52 < npou> one easy option is to say that the Eq instance will be slow, (==) = (==) `on` 
              (`appendUriParts` [])
```

**Comment by fisx** [comment]
I am about to commit this in `Frontend.Path`:

```haskell
main (Login mb)       root = root </> (fromString ("login" <> if fromMaybe True mb then "" else "?status=False"))
```

it works, but it's certainly not as nice as uri-bytestring.  so when deciding where to go with `Data.UriPath`, we should thing about queries.

**Comment by fisx** [comment]
Last comment has been resolved in #406.  Closing for now, a clear need for uri-bytestring does not appear to be emerging any time soon.
